### DISCUSSION TITLE
invalidate does not work well in production mode

### DISCUSSION THREAD

**Discussion by yuanxintaxon** [TanStack/query#2891] [Q&A] [answered]
invalidates does not work well in production mode. It only works if i do window focus fetch.

**Comment by TkDodo** [comment, marked as answer]
you have to be more specific than that please. Do you have an example codesandox?

**Reply by yuanxintaxon** [reply]
![image](https://user-images.githubusercontent.com/22881271/140627098-bc1109f0-a0c2-4a3c-b308-f082bd9bf156.png)
When i delete a forum thread, it should invalidate the query on success. But it seems like, they invalidate the query before the deletion request arrive at the backend server

**Reply by yuanxintaxon** [reply]
![image](https://user-images.githubusercontent.com/22881271/140627131-74198382-c2a6-48e1-99a1-29205b071718.png)
I added one alert() and that slows down the invalidation request, making it work.

**Reply by yuanxintaxon** [reply]
![image](https://user-images.githubusercontent.com/22881271/140627176-53ac10e0-3f3a-4449-bd03-396160ec5537.png)
This should be the expected order. PUT > GET. (the issue i am having is the invalidation request was invoked before the the PUT method request reaches my server)

**Reply by yuanxintaxon** [reply]
May i know is there a way to solve this? Will be glad if someone is kind to help

**Reply by yuanxintaxon** [reply]
My current workaround is to setTimeOut 500 seconds to delay the invalidation. But that's too weird.

**Reply by dusty** [reply]
Don't you need to return the promise?

```
return fetchJson()...
```

**Reply by yuanxintaxon** [reply]
Oh man!! thank you so much. didnt spot that!!  you are my life saver ^ ^

**Reply by dusty** [reply]
np. 

A second set of eyes is always helpful - been there thousands of times.  :)
